kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -246,13 +255,19 @@ private long commitWithRetry(
       // AddFile actions that do not yet have them. If the row ID high watermark changes, emit a
       // DomainMetadata action to update it.
       if (TableFeatures.isRowTrackingSupported(protocol)) {
+        if (providedRowIdHighWatermark.isPresent()) {
+          // Conflict resolution is disabled when providedRowIdHighWatermark is set,
+          // because it must be updated according to the latest table state.
+          maxRetries = 0;
```

Comment:
Can we just throw? It is a bad user experience.

Reply:
Can we do it at set time e.g.

https://github.com/delta-io/delta/pull/4856/files#diff-e18199e992a0835a88ce135292bb9d3b6b7cc004406108d6eefd5981bdd06354R846